purchase_request/models/purchase_order.py

```diff
@@ -214,7 +214,7 @@ def _purchase_request_confirm_done_message_content(self, message_data):
             "product_uom": message_data["product_uom"],
         }
         message += "</ul>"
-        return message
+        return Markup(message)
```

Comment:
While you are fixing the treatment of html in this module, can you apply `odoo.tools.misc.html_escape` to all string values in message_data above? And the same in the other files.

Reply:
Sorry, I don't understand the need to use html_escape, can you explain a use case (for understanding) where it is necessary to use Markup + html_escape?

Reply:
When you cast text as Markup, you are claiming that its contents are valid and safe HTML. Therefore, you need to escape all the raw strings you insert into it.
![image](https://github.com/user-attachments/assets/127c1f45-5448-4ebf-bbde-051801fdaf4b)
![image](https://github.com/user-attachments/assets/35b243d5-d9db-4565-add3-83a61b9d2e12)

Reply:
But, do we need to control that? I think we should not, there are many sites in odoo core where they use Markup() and not html_escape, if for example there is “incorrect html code” in the partner name the same thing would also happen, so IMO if there is an incorrect code somewhere (in the example you indicate in the product name) what should be done is to correct the name, but it is just my opinion.

Reply:
Do you have an example where a partner name is inserted in a HTML markup message without escaping?

Reply:
For the record, I'm calling it 'escape' because the method for this provided by Odoo is `html_escape` which is an alias for Markupsafe.escape: https://github.com/odoo/odoo/blob/18.0/odoo/tools/misc.py#L1288

Reply:
BTW another option is to use Markup formatting, which applies escaping itself: https://pypi.org/project/MarkupSafe/. You can then concatenate the Markup safe fragments that combine the bullet list in a message.

Reply:
`html2text` is also a method in Odoo in `tools`. I didn't know about the `Markupsafe.escape` before.